### PR TITLE
Make 'HttpResponseContext' not needed in PS function script

### PR DESCRIPTION
**Pull request by daxian-dbw**
Make `HttpResponseContext` not needed in PS function script.

To properly convert a value for a specific output binding name implicitly, we need to know the metadata of the binding referred by that name, such as the type of the output binding.
The changes in this PR register the output binding metadata to a concurrent dictionary with the Runspace's InstanceId as the key before start running a function script, and then un-register after the function script finishes running. When a function script that calls `Push-OutputBinding` is running, we can figure out the current Runspace by `Runspace.DefaultRunspace`, and then we can query for the output binding metadata by using the InstanceId.

This approach take into the concurrency support that we will have in future. `Push-OutputBinding` is able to get the metadata about the function script that is running even with multiple Runspaces/PowerShellManagers processing requests at the same time.

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
Looks like there's a CI failure. I think you missed including the Resources file?

File: src/Modules/Microsoft.Azure.Functions.PowerShellWorker/Microsoft.Azure.Functions.PowerShellWorker.psm1 [hunk 1]
```diff
@@ -49,6 +55,52 @@ function Get-OutputBinding {
     }
 }
 
+# Helper private function that validates the output value and do necessary conversion.
```
[the remark this answers is not in the document]
Comment (daxian-dbw) [reply src/Modules/Microsoft.Azure.Functions.PowerShellWorker/Microsoft.Azure.Functions.PowerShellWorker.psm1]: Fixed.

File: src/Modules/Microsoft.Azure.Functions.PowerShellWorker/Microsoft.Azure.Functions.PowerShellWorker.psm1 [hunk 2]
```diff
@@ -49,6 +55,52 @@ function Get-OutputBinding {
...
+    )
+
+    # Check if we can get the binding metadata of the current running function.
+    $funcMetadataType = "FunctionMetadata" -as [type]
+    if ($null -eq $funcMetadataType) {
+        return $Value
+    }
+
+    # Get the runspace where we are currently running in.
+    $runspace = [Runspace]::DefaultRunspace
```
[the remark this answers is not in the document]
Comment (daxian-dbw) [reply src/Modules/Microsoft.Azure.Functions.PowerShellWorker/Microsoft.Azure.Functions.PowerShellWorker.psm1]: Updated.

**Comment by daxian-dbw** [comment]
It turns out the `.gitignore` cause the `resource.psd1` file to be ignored.

**Comment by daxian-dbw** [comment]
> I was wondering if we could hold off on coupling the `FunctionLoader` and the `PowerShellManager` just to keep the separation of concerns but I can't think of a way to keep it that way.

@TylerLeonhardt and I chatted about this offline. This PR doesn't introduce coupling between `FunctionLoader` and `PowerShellManager`. `FunctionMetadata` serves as a store that holds the function metadata information for the function script that is running in each `Runspace` (query key is the Runspace's InstanceId). The function metadata is registered before a `PowerShellManager` runs the corresponding function script, and is unregistered after the corresponding function script is done running.

**Pull request merged**